### PR TITLE
colour: improve ICC compatibility check for CMYK images

### PR DESCRIPTION
**Pull request by kleisauke**
See: #3564.

Targets the 8.14 branch.

### CODE REVIEW

File: libvips/colour/icc_transform.c [hunk 1]
```diff
@@ -575,22 +570,24 @@ vips_image_expected_bands( VipsImage *image )
...
 
 		case VIPS_INTERPRETATION_MULTIBAND:
 		case VIPS_INTERPRETATION_HISTOGRAM:
-		case VIPS_INTERPRETATION_CMYK:
 		case VIPS_INTERPRETATION_MATRIX:
 		case VIPS_INTERPRETATION_FOURIER:
 		default:
-			expected_bands = image->Bands;
-			break;
+			return( image->Bands == profile_bands );
```
Comment (jcupitt) [inline libvips/colour/icc_transform.c:589]: Maybe ">="? There can be more image bands than profile bands if there are alpha channels.

Comment (kleisauke) [reply libvips/colour/icc_transform.c]: Commit bec4942c6af0e629857da158c7e8657dfa0e1fff relaxes this check to `>=`.

**Comment by jcupitt** [comment]
This looks too easy! I can't help feeling there's some corner case we've overlooked, but I don't know what it is.

**Comment by kleisauke** [comment]
As well as checking it against the test case in #3564, I also checked it against issue #3139 and the test cases in PR #3372. But indeed, perhaps we overlooked something, I will verify some old issues.

**Comment by jcupitt** [comment]
I checked it against CMYKs with sRGB profiles, RGBs with CMYK profiles, and mono images with RGB profiles, and it all seems fine. I didn't try alpha channels.

I think maybe I'm only uneasy because I remember changing this logic before and breaking everything :( It's probably fine.

**Comment by kleisauke** [comment]
The test case mentioned in https://github.com/libvips/libvips/issues/730#issuecomment-1345375891 that was fixed with PR #3139 also still produces the expected behavior.

I'll do some more testing with CMYK images, since that's the only behavior change in this PR (the change in commit bec4942c6af0e629857da158c7e8657dfa0e1fff is also a behavior change, but that's much more uncommon, I think).
